runtime/lua/vim/lsp/handlers.lua

```diff
@@ -215,9 +215,14 @@ M['textDocument/documentSymbol'] = symbol_handler
 M['workspace/symbol'] = symbol_handler
 
 --@see https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_rename
-M['textDocument/rename'] = function(_, _, result)
-  if not result then return end
-  util.apply_workspace_edit(result)
+M['textDocument/rename'] = function(_, _, result, client_id)
+  if not result or #result.documentChanges == 0 then
```

Comment:
It looks like the response is a [WorkspaceEdit](https://microsoft.github.io/language-server-protocol/specifications/specification-current/#workspaceEdit) so `documentChanges` could be `nil` and instead it could also have `changes` set

Reply:
Ahh you're totally right. Thanks and good catch! I changed this around a bit and added in a util function to help out with determining this, and also added some tests for it.